src/app/3d/qgs3dmaptoolmeasureline.cpp

```diff
@@ -52,11 +52,20 @@ Qgs3DMapToolMeasureLine::Qgs3DMapToolMeasureLine( Qgs3DMapCanvas *canvas )
 {
   mPickHandler.reset( new Qgs3DMapToolMeasureLinePickHandler( this ) );
 
-  // Dialog
-  mDialog = new Qgs3DMeasureDialog( this );
-  mDialog->setWindowFlags( mDialog->windowFlags() | Qt::Tool );
-  mDialog->restorePosition();
+  if ( mDialog )
+    activate();
```

Comment:
I don't understand why this change is necessary - mDialog should be always nullptr in the constructor at this point...

Reply:
I thought that since the dialog was deactivated the aim was to reactivate it at some point. I'm not sure if by always deactivating we end up with plenty of hidden dialogs or if it get cleaned up at some point.

And so I thought that by moving to deactivate instead of a destructor was to reuse the dialog and I thought that this was the way to do so.